### PR TITLE
CDAP-19098: react migration of Studio toppanel Name/Description

### PR DESCRIPTION
**Pull request by de-lan**
# React migration of Studio toppanel Name/Description

## Description
This PR is targeted to a feature branch of the the story: migration from AngularJS to React on Studio Toppanel

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [x] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-19098](https://cdap.atlassian.net/browse/CDAP-19098)

## Test Plan
Manual

## Screenshots
image

### CODE REVIEW

**Comment by seanfeld** [comment]
changed the branch to be something easier to remember

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    color: #d15668;
+  `}
+`;
+
+const PipelineDescription = styled.div`
+  color: gray;
+  font-size: 12px;
+`;
+
+const useStyle = makeStyle(() => {
```
Comment (seanfeld) [inline app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx]: Couldn't use styled components with these?

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            value={name}
+          />
+          <textarea
+            value={description}
+            placeholder="Enter a description for your pipeline."
+            data-cy="pipeline-description-input"
+            onChange={handleDescriptionChange}
+          ></textarea>
+          <div className="btn-group pull-left">
+            <button
```
Comment (seanfeld) [inline app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx]: Use the material ui button

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          >
+            <PipelineDescription>
+              {parsedDescription || 'Enter a description for your pipeline'}
+            </PipelineDescription>
+          </Tooltip>
+        </div>
+      )}
+      {metadataExpanded && (
+        <div>
+          <input
```
Comment (seanfeld) [inline app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx:113]: use material ui versions of all these html elements

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            type="text"
+            id="pipeline-name-input"
+            placeholder="Name your pipeline"
+            onKeyUp={onEnterOnMetadata}
+            onChange={handleNameChange}
+            value={name}
+          />
+          <textarea
```
Comment (seanfeld) [inline app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx:121]: here too

**Review by seanfeld — CHANGES_REQUESTED** [review]
Great work so far! We want to update these things to use material ui standard stuff instead of the old bootstrap styling - we'll do a design review later

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx (hunk 2 above)
Comment (de-lan) [reply app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx]: yeah, will update

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx (hunk 3 above)
Comment (de-lan) [reply app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx:113]: After some trying, found that the input and textarea styles are tangled deeply in the toppanel bootstrap styling, decided to leave it for now, and will address them all together after the whole top panel is migrated.

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx (hunk 4 above)
Comment (de-lan) [reply app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx:121]: same above

File: app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx (hunk 1 above)
Comment (de-lan) [reply app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx]: updated

**Review by seanfeld — APPROVED** [review]

**Pull request merged**